File: TaskProject/ClientApp/src/app/shared/Task.ts
import { Member } from "./member";

export class Task {
    Task_Id :number;
    FK_Member_Id :number;
    Description :string;
    deliver_date:string;
    IsCompleted:boolean;
    IsDeleted:boolean;
    Member?:any;
    constructor(_Task_Id: number,_FK_Member_Id:number, _Description :string,_deliver_date :string,_IsCompleted:boolean,_IsDeleted:boolean,_Member:any) {
        this.Task_Id = _Task_Id;
        this.FK_Member_Id = _FK_Member_Id;
        this.Description = _Description;
        this.IsCompleted = _IsCompleted;
        this.deliver_date=_deliver_date;
        this.IsDeleted=_IsDeleted;
        this.Member=_Member;
}
}

File: TaskCore/Models/Task.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TaskCore.Models
{
   public class Task
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Task_Id { get; set; }
        [Required, ForeignKey("Member")]
        public int FK_Member_Id { get; set; }
        // public string MemberName { get; set; }
        [Required, MaxLength(100), MinLength(10)]
        public string Description { get; set; }
        [Required]
        public string deliver_date { get; set; }
        public bool IsCompleted { get; set; }
        public bool IsDeleted { get; set; }
        public virtual Member Member { get; set; }
    }
}
